File: frontend/lib/models/light_type.dart
enum LightType {
  led('Led', 'LED'),
  sodium('Sodyum', 'SODIUM');

  final String displayName;
  final String apiValue;

  const LightType(this.displayName, this.apiValue);

  static LightType fromApiValue(String value) {
    return LightType.values.firstWhere(
      (e) => e.apiValue == value,
      orElse: () => throw ArgumentError('Bilinmeyen Cisim Türü: $value'),
    );
  }
}

File: backend/cografi-nesne-takip/src/main/java/com/kutalmis/cografi_nesne_takip/entity/LightType.java
package com.kutalmis.cografi_nesne_takip.entity;

public enum LightType {
    LED("Led"),
    SODIUM("Sodyum");

    private final String displayName;

    LightType(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }
}